src/testdir/test_filetype.vim

```diff
@@ -2340,4 +2341,30 @@ func Test_vba_file()
   filetype off
 endfunc
 
+func Test_swig_file()
+  filetype on
+
+  " Swig
+
+  call writefile(['%module mymodule'], 'Xfile.i', 'D')
+  split Xfile.i
+  call assert_equal('swig', &filetype)
+  bwipe!
+
+  " Swig
+
+  call writefile(['%{', '#include <header.hpp>", '%}'], 'Xfile.i', 'D')
```

Comment:
```suggestion
  call writefile(['%{', '#include <header.hpp>', '%}'], 'Xfile.i', 'D')
```

Reply:
That still doesn't suffice though

```
1 FAILED:
Found errors in Test_swig_file():
command line..script /home/julien/Software/Others/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_swig_file line 7: Expected 'swig' but got 'progress'
command line..script /home/julien/Software/Others/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_swig_file line 20: Expected 'asm' but got 'progress'
```

Reply:
Ok, I fixed it locally. @mcepl Are you planning on wrapping this PR? Otherwise I should open another PR, because I don't have rights to write to your branch and this is going to get annoying.

Reply:
See [my other comment](https://github.com/vim/vim/pull/13540#discussion_r1403268413) … if you are willing to take over, then you are more than welcome. Just please leave there at least one commit from me.